migration/migrator/migrations/master/20240219141515_docker_image.py

```diff
@@ -0,0 +1,45 @@
+"""Migration for the Submitty master database."""
+import json
+
+def up(config, database):
+    """
+    Run up migration.
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    """
+    database.execute("""
+        CREATE TABLE IF NOT EXISTS public.docker_images (
+            image_name character varying UNIQUE NOT NULL,
+            user_id character varying REFERENCES users (user_id)
+        );
+    """)
+
+    # Add existing containers to db with no owner
+    container_file_path = config.config_path / 'autograding_containers.json'
+    with open(container_file_path) as container_file:
+        container_data = json.load(container_file)
+    images = set()
+    for capability in container_data:
+        for image in container_data[capability]:
+            images.add(image)
+    existing_images = set()
+    for row in database.execute("SELECT image_name FROM docker_images").all():
+        existing_images.add(row[0])
+    for image in images:
+        if image not in existing_images:
+            database.session.execute("INSERT INTO docker_images (image_name, user_id) VALUES (:name, NULL);", {"name": image})
```

Comment:
Thoughts on setting user_id to the submitty admin account rather than just null? This way we can enforce a non-null constraint.

Reply:
Is there an admin user that is common across production and development environments? If so I can add that but I don't want to assign a foreign key column a default value that might not be present.